lua/autorun/nl_library.lua

```diff
@@ -0,0 +1,476 @@
+		ply:TakeDamageInfo(dmginfo) // Apply all ragdoll damage directly to the player.
+		ply.ragdolldamage = false
+	end
+end)
+
+hook.Add( "PlayerUse", "NLLRagdollWakeAttempt", function( ply, ent )
+    if IsValid(ent) and IsValid(ent.nllplayer) then //Don't wake players that don't have their condition set to "ZZZ"
+        if ent.nllplayer.condition != "ZZZ" and not ply:IsWalking() then return end // allow 'waking' if the victim is stunned and needs to be killed
+        plytarg = ent.nllplayer
+        NLL.UnRagdoll(plytarg)
+	plytarg.nllismuted = false
+        if timer.Exists("NLLUnragdoll".. plytarg:UserID()) then
+            timer.Remove( "NLLUnragdoll".. plytarg:UserID())
+        end
+        if not SERVER then return end
+        
+	if ply:IsWalking() then // alt + e to kill instead of waking
+	    if ply.CovertBuddha == nil then ply.CovertBuddha = 0 end
+	    if ply.CovertBuddha == 1 then
+		plytarg:SetHealth(1)
+		return
+	    end
+            local dmgtable = DamageInfo()
+            dmgtable:SetDamage((plytarg:Health())*1.5)
+            dmgtable:SetAttacker(ply)
+            dmgtable:SetDamageForce(Vector(0,0,-1))
+            dmgtable:SetDamagePosition(plytarg:GetPos())
+            dmgtable:SetDamageType(4)
+            dmgtable:SetInflictor(ply)
+            timer.Create("NLLKillTarget".. plytarg:UserID(), 0.2, 1, function()
+                plytarg:TakeDamageInfo( dmgtable )
+            end)
+        end
+    end
+end)
+	
+
+function NLL.PlayerZZZ( ply, pushdir, timebefore, timedur )
+	local id = ply:UserID()
+        
+	if ply.isDrowsy then return end // don't fuck with drowsy players
+	if timer.Exists("NLLSleepIn"..id) then
+		if timer.TimeLeft( "NLLSleepIn"..id ) < 3 then return end
+		local newdelay = math.min( timebefore/3, timer.TimeLeft( "NLLSleepIn"..id )/2 )
+		timer.Remove("NLLSleepIn"..id)
+		timer.Create("NLLSleepIn"..id, newdelay, 1, function()
+			NLL.Ragdoll(ply, pushdir,"ZZZ")
+			ply.nllismuted = true
+			timer.Create("NLLUnragdoll"..id, timedur, 1, function()
+				NLL.UnRagdoll(ply)
+				ply.nllismuted = false
+			end)
+		end)
+		return
+	end
+	timer.Create("NLLSleepIn"..id, timebefore, 1, function()
+		NLL.Ragdoll(ply, pushdir,"ZZZ")
+		ply.nllismuted = true
+		timer.Create("NLLUnragdoll"..id, timedur, 1, function()
+			NLL.UnRagdoll(ply)
+			ply.nllismuted = false
+		end)
+	end)	
+end
+
+if CLIENT then
+
+function NLL_BuddhaKillUpdate(ply,cmd,args)
+	if args[1] != "0" then
+		net.Start("nllupdatebuddha")
+		net.WriteInt(1,4)
+		net.SendToServer()
+	else
+		net.Start("nllupdatebuddha")
+		net.WriteInt(0,4)
+		net.SendToServer()
+	end
+end
+
+concommand.Add( "ttt_nl_buddhakill", function(ply,cmd,args) NLL_BuddhaKillUpdate(ply,cmd,args) end, nil, "If nonzero, covertly killing will instead set your victim's hp to 1", 0 )
+
+net.Receive("nllstartview", function()
+	local rag = net.ReadEntity()
+	LocalPlayer().nllviewrag = rag
+end)
+net.Receive("nllendview", function()
+	LocalPlayer().nllviewrag = nil
+end)
```

Comment:
These two net messages could be consolidated into one, or just turned into a NWEntity variable